dissect/target/plugins/os/unix/commandhistory.py

```diff
@@ -5,34 +5,54 @@
 from dissect.target.helpers.record import create_extended_descriptor
 from dissect.target.helpers.descriptor_extensions import UserRecordDescriptorExtension
 
-BashHistoryRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
-    "linux/bashhistory",
+CommandHistoryRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
+    "linux/history",
     [
         ("datetime", "ts"),
         ("wstring", "command"),
         ("uri", "source"),
     ],
 )
 
+COMMAND_HISTORY_FILES = [".bash_history", ".zsh_history", ".fish_history", "fish_history", ".python_history"]
+IGNORED_HOMES = ["/bin", "/usr/sbin", "/sbin"]
 
-class BashHistoryPlugin(Plugin):
+
+class CommandHistoryPlugin(Plugin):
     def check_compatible(self):
-        pass
+        for user_details in self.target.user_details.all_with_home():
+            for file_ in user_details.home_path.iterdir():
+                if file_.name in COMMAND_HISTORY_FILES:
+                    return True
+        return False
 
-    @export(record=BashHistoryRecord)
+    @export(record=CommandHistoryRecord)
     def bashhistory(self):
-        """Return bash history for all users.
+        """
+        Deprecated, use commandhistory function.
+        """
+        return self.commandhistory()
 
-        When using the BASH shell, history of the used commands is kept on the system. It is kept in a hidden file
-        named ".bash_history" and may expose commands that were used by an adversary.
+    @export(record=CommandHistoryRecord)
+    def commandhistory(self):
+        """Return shell history for all users.
+
+        When using a shell, history of the used commands is kept on the system. It is kept in a hidden file
+        named ".$SHELL_history" and may expose commands that were used by an adversary.
         """
+
         for user_details in self.target.user_details.all_with_home():
+
+            for ih in IGNORED_HOMES:
+                if ih in user_details.user.home:
+                    continue
+
             for file_ in user_details.home_path.iterdir():
-                if not file_.name.startswith(".bash_history"):
+                if file_.name not in COMMAND_HISTORY_FILES:
                     continue
 
                 try:
-                    for line in file_.open("rt"):
+                    for line in file_.open("rt", errors="ignore"):  # Ignore Non-ASCII characters in bash_history
```

Comment:
```suggestion
                    for line in file_.open("rt", errors="ignore"):  # Ignore Non-UTF-8 characters in bash_history
```

Wouldn't it be better to use `errors="replace"` so you know you are potentially missing something?